### PR TITLE
10 feature infobox for resources

### PR DESCRIPTION
**Pull request by NilsArnlund**
- Added backend functions to get info-box data to front end.
- Added info-box component
- Added test for info-box

NOTE:
- This has already been merged with #21 Tabs to save some time. (This will also be accepted into main first.)
- The addition of infoboxes makes the problem of navbar covering tabs even more apparent. This will be fixed in another issue, #85 and therefore is not addressed here.
- The info boxes data is current a given value and not coming from an uploaded file. This will be implemented in #79

### CODE REVIEW

**Comment by AlgotE** [comment]
The code looks good but it seems like it has infoboxes hardcoded and these are not visible for me. Is this a draft or is it meant to be implemented later in #79?

bild

**Comment by AlgotE** [comment]
I got it to show the infoboxes, it was not clear that the files had to be "rtx" && "rta".

**Comment by NilsArnlund** [comment]
Hmm I get it to crash in Chrome as well, but only when uploading a single reasource file. Uploading multiple files does not result in a crash. I tried in Firefox as well and got the same result. Uploading, for exemple rta.json and 4b14a8_230206.txt followed by another file to update Firefox works. I am current unsure why this is, I will look into it, but I am almost certain it has to do with they way the info-boxes are "hard coded" and would the be solved in #79

**Comment by NilsArnlund** [comment]
> I got it to show the infoboxes, it was not clear that the files had to be "rtx" && "rta".

I think this is it. They way the info-boxes are "hard coded" make it so that any other file crashes. @Dethcrvsh could you also take a look at this?

**Review by ViktorAndersson99 — CHANGES_REQUESTED** [review]

File: src/components/Charts.tsx [hunk 1]
```diff
@@ -133,15 +138,19 @@ function getCandleChartData(
...
+    if (fileContent !== '{}') {
+      const data = JSON.parse(fileContent)[metric];
+      candle.candles.push({
+        x: index + 1,
+        open: data.first_quartile,
+        close: data.third_quartile,
+        high: data.max,
+        low: data.min,
+      });
+      // TODO: Implement mean
```
[the remark this answers is not in the document]
Comment (ViktorAndersson99) [reply src/components/Charts.tsx:152]: I might be missing something. Also this does not have to be fixed now, just my other comment

[the remark this answers is not in the document]
Comment (NilsArnlund) [reply src/components/Charts.tsx:152]: This won't be the final implementation. This is changed in #84

File: src/components/ViewTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import Tabs from '@mui/material/Tabs';
+import Tab from '@mui/material/Tab';
+import Box from '@mui/material/Box';
+
+import { styled } from '@mui/material/styles';
+
+import { BoxGraphComponent, BarGraphComponent } from './BaseComponent';
+
+interface ViewTabsProps {
+  siteProps: any;
```
[the remark this answers is not in the document]
Comment (NilsArnlund) [reply src/components/ViewTabs.tsx:15]: This should not be fixed in this PR. Should instead be commented on in  PR #75

**Review by ViktorAndersson99 — APPROVED** [review]

**Pull request merged**